lms/djangoapps/course_blocks/transformers/hidden_content.py

```diff
@@ -97,6 +101,10 @@ def _is_block_hidden(self, block_structure, block_key):
         Returns whether the block with the given block_key should
         be hidden, given the current time.
         """
-        due = self._get_merged_due_date(block_structure, block_key)
         hide_after_due = self._get_merged_hide_after_due(block_structure, block_key)
-        return not SequenceModule.verify_current_content_visibility(due, hide_after_due)
+        self_paced = block_structure[block_structure.root_block_usage_key].self_paced
+        if self_paced:
+            date = block_structure[block_structure.root_block_usage_key].end
```

Comment:
Nit: Consider renaming this variable to `hidden_date` to be consistent with the improved naming chosen elsewhere.